test/Microsoft.AspNetCore.Mvc.FunctionalTests/Infrastructure/HttpClientExtensions.cs

```diff
@@ -32,7 +31,7 @@ public static class HttpClientExtensions
 
         public static async Task<HttpResponseMessage> AssertStatusCodeAsync(this HttpResponseMessage response, HttpStatusCode expectedStatusCode)
         {
-            if (response.StatusCode == HttpStatusCode.OK)
+            if (response.StatusCode == expectedStatusCode)
```

Comment:
As above, if anyone objects to this test cleanup in a patch, I'll revert changes to this file.